libraries/GCS_MAVLink/GCS_Common.cpp

```diff
@@ -5169,6 +5169,30 @@ MAV_RESULT GCS_MAVLINK::handle_do_set_safety_switch_state(const mavlink_command_
     }
 }
 
+MAV_RESULT GCS_MAVLINK::handle_do_set_sys_cmp_id(const mavlink_command_int_t &packet, const mavlink_message_t &msg)
+{
+    const uint8_t new_system_id = packet.param1;
+    const uint8_t new_component_id = packet.param2;
+    const uint8_t do_reboot = packet.param3;
+
+    if (new_system_id == 0) {
+        return MAV_RESULT_FAILED;
+    }
+
+    if (!AP_Param::set_and_save_by_name("SYSID_THISMAV", new_system_id)) {
+        return MAV_RESULT_FAILED;
+    }
+
+    if (new_component_id != 0 && !AP_Param::set_and_save_by_name("COMPID_THISMAV", new_component_id)) {
+        return MAV_RESULT_FAILED;
+    }
+
+    if (do_reboot) {
+        hal.scheduler->reboot(false);
```

Comment:
If this returns then there's been a failure, so we should not return ACCEPTED.

Reply:
What should we return ?

Reply:
`MAV_RESULT_FAILED` looks appropriate.

Reply:
I'm not sure if I follow you logic here. I just moved the reboot action to a late_reboot. The idea is to allow the vehicle to ACK the command before rebooting.